src/executor/sort_agg.rs

```diff
@@ -0,0 +1,189 @@
+                if last_key.len() != 0 && group_key.eq(&last_key) {
+                    let states = states_vec.last_mut().unwrap();
+
+                    for (state, expr) in states.iter_mut().zip_eq(&exprs) {
+                        state.update_single(&expr.get(row_idx))?;
+                    }
+                } else {
+                    let mut states = create_agg_states(&self.agg_calls);
+                    for (state, expr) in states.iter_mut().zip_eq(&exprs) {
+                        state.update_single(&expr.get(row_idx))?;
+                    }
+                    states_vec.push(states);
+                    last_key = group_key;
+                }
+            }
+        }
+
+        #[for_await]
+        for chunk in Self::finish_agg(states_vec) {
+            let chunk = chunk?;
+            yield chunk
+        }
+    }
+
+    #[try_stream(boxed, ok = DataChunk, error = ExecutorError)]
+    async fn finish_agg(states_vec: Vec<SmallVec<[Box<dyn AggregationState>; 16]>>) {
+        for ele in states_vec {
+            yield ele
+                .iter()
+                .map(|s| {
+                    let result = &s.output();
+                    match &result.data_type() {
+                        Some(r) => {
+                            let mut builder = ArrayBuilderImpl::with_capacity(1, r);
+                            builder.push(result);
+                            builder.finish()
+                        }
+                        None => ArrayBuilderImpl::new(&DataTypeKind::Int(None).nullable()).finish(),
+                    }
+                })
+                .collect::<DataChunk>()
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+
+    use super::*;
+    use crate::array::ArrayImpl;
+    use crate::binder::{AggKind, BoundInputRef};
+    use crate::types::{DataType, DataTypeKind};
+
+    #[tokio::test]
+    async fn test_multi_group_agg() {
+        let agg_calls = vec![create_sum_agg_call(0), create_sum_agg_call(1)];
+
+        let group_keys = vec![create_input_ref(1), create_input_ref(2)];
+
+        let col1 = vec![1.1, 0.2, 0.3, 0.4, 0.5];
+        let col2 = vec![1.1, 1.1, 1.3, 1.4, 1.5];
+        let col3 = vec![1.2, 1.2, 2.3, 2.4, 2.5];
+        let child: BoxedExecutor = async_stream::try_stream! {
+            yield  vec![ArrayImpl::new_float64(col1.into_iter().collect()),ArrayImpl::new_float64(col2.into_iter().collect()),ArrayImpl::new_float64(col3.into_iter().collect())]
+            .into_iter().collect()
+        }.boxed();
+
+        let executor = SortAggExecutor {
+            agg_calls,
+            group_keys,
+            child,
+        };
+        let mut executor = executor.execute();
+
+        let expected_col = vec![1.3, 2.2, 0.3, 1.3, 0.4, 1.4, 0.5, 1.5];
+        let mut result_col = Vec::new();
+        while let Some(chunk) = executor.next().await {
+            for chunk in chunk.unwrap().arrays() {
+                result_col.push(chunk.clone());
+            }
+        }
```

Comment:
This test code seems a little bit confusing. Suggest to represent `expected` as `vec![vec![1.3, 2.2], ...]` And then assert each chunk looks like that.